core/domain/html_validation_service.py

```diff
@@ -712,6 +713,32 @@ def add_caption_attr_to_image(html_string):
     return python_utils.UNICODE(soup)
 
 
+def validate_url_with_value_for_links(html_list):
+    """Validates url-with-value field in link component in a list of
+    html string.
+
+    Args:
+        html_list: list(str). List of html strings to be validated.
+
+    Returns:
+        list(str): List of all the html strings where the link component is
+            invalid.
+    """
+    err_list = []
+
+    for html_string in html_list:
+        soup = bs4.BeautifulSoup(
+            html_string.encode(encoding='utf-8'), 'html.parser')
+
+        for link in soup.findAll(name='oppia-noninteractive-link'):
+            if (not link.has_attr('url-with-value') or
+                    link['url-with-value'] == '' or
+                    link['url-with-value'] == '&quot;&quot;' or
+                    re.match(r'.*mailto:.*', link['url-with-value'])):
+                err_list.append(python_utils.UNICODE(link))
```

Comment:
Don't you want to get the info for which exploration and state it's in, etc., as well?

I think you'll want the counts of each type of error as well, so consider grouping them by key when you pass them to the reduce() method in the job. That way you can associate each type of error to a list of instances and get a sense of the relative prevalence of the errors. Try to get as much useful information as possible from running the job just once, so that we don't need a lot of iteration.

Reply:
Done. PTAL @seanlip!